.github/workflows/test.yml

```
@@ -8,12 +8,12 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout Repo
-        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+        uses: actions/checkout@v3
```

```suggestion
        uses: actions/checkout@dcd71f646680f2efd8db4afa5ad64fdcba30e748
```

I think Stephen had a reason to pin to SHA rather than version numbers. Also 3.0.1 [released](https://github.com/actions/checkout/releases) a few days ago.

---

Tags can be changed, which means it's an avenue for remote code execution (They push a new commit with some malicious code and then change the tag so we get the new version and execute it automatically). It's unlikely for GitHub to be compromised like that but it's definitely possible with other repositories, so it's safer to use a SHA for all workflows.